### ISSUE TITLE
[TF2] Error when using only intercept values for design_scale

### ISSUE THREAD

**Issue by kadam0** [theislab/batchglm#96]
Generating simulated data given the following parameters:

- 100 genes, 10,000 cells, 4 conditions + 2 batches each + numeric size factor covaritate, ~1 dispersion model

We generated the following data matrices:

- design loc, shape:  (10000, 5)
- design scale, shape:  (10000, 1)
- x, shape:  (10000, 100)
- param matrices, shapes:  (5, 100) (1, 100)

To try to fit the nb-model, we used that simulated data to create an `glm.models.tf2.glm_nb.InputDataGLM` object:
```
input_data = InputDataGLM(
                data=x,
                design_loc=design_loc,
                design_scale=design_scale,
                as_dask=False
          )
```
In the InputDataGLM class, the constraints are created:

https://github.com/theislab/batchglm/blob/e4a5594b821ce2163c04fe59b0b7fd46354d2b5b/batchglm/models/base_glm/input.py#L118-L129

While the `constraints_loc` is created without problems using our simulated data:
```
[[1. 0. 0. 0. 0.]
 [1. 1. 0. 0. 0.]
 [1. 0. 1. 0. 0.]
 ...
 [1. 1. 0. 0. 1.]
 [1. 0. 1. 0. 1.]
 [1. 0. 0. 1. 1.]]
shape: (10000, 5)
```
We get an error, when the `constraints_scale` is created:
```
.../batchglm/models/base_glm/utils.py", line 68, in parse_constraints
    constraints = np.identity(n=dmat.shape[1])
IndexError: tuple index out of range
```
The problem seems to be in this code:
https://github.com/theislab/batchglm/blob/e4a5594b821ce2163c04fe59b0b7fd46354d2b5b/batchglm/models/base_glm/utils.py#L65-L67

Given, that the output of the `design_scale` is as follows, as it only has the intercept:
```
[1. 1. 1. ... 1. 1. 1.]
shape: (10000,)
```
the problems seems clear, as there is not `(10000,1)` as shape, as you would expect it, and therefore `dmat.shape[1]` does not exist.

How can we resolve this problem?

**Comment by picciama** [comment]
The problem to me seems somewhere else:
In the beginning, you wirte the shape of design scale is (10000,1), so `dmat.shape[1] = 1`, but in the end the sceond dimension is lost. 
Somewhere between must be the error, which can happen when reading the matrix columnwise. In this case, it is only one column so the result is a vector. 
You need to expand to a second dimension to circumvent the problem:
```np.expand_dims(dmat, axis=1)```

**Comment by picciama** [comment]
Your problem was related to reading in the design matrix from disk.
I checked the file and it is a csv with only one column, which is why 
```np.genfromtxt(dscale.csv, delimiter=',')```
This function  uses the line break as indication to add the next value in the direction of the first dimension. Values separated by the delimiter are appended in the direction of the second dimension, which is not expanded because it is only one value. 
I introduced a check in the script you are using to read in the data:
```
design_scale = np.genfromtxt(in_dir+'dscale.csv', delimiter=',')
    if design_scale.ndim == 1:
        design_scale = np.expand_dims(design_scale, axis=1)
```
It is working now.
